### PR TITLE
Feat: added terraform state bucket validation

### LINKED ISSUE #137: [Furyctl Next] introduce check for terraform state bucket in `furyctl validate dependencies`
Furyctl Next shall check if the bucket configured for storing the terraform state exists when running the `validate dependencies` command.

### LINKED ISSUE #219: Please specify which `awscli` version is required
When download dependencies, please specify which `awscli` version is required.

In my case, I have already installed to a newer version and it's not compatible with the KFD requirements.

Please, provide an automatic and indipendent way to download `awscli` when download dependencies

### PR DESCRIPTION
**Pull request by Al-Pragliola**
Changelist:

- added a step in the validation process, to check if the bucket from the furyctl.yaml exists
- added terraform semver constraint support

needs https://github.com/sighupio/fury-distribution/pull/51

partially fixes #219 

closes #137

### CODE REVIEW

File: internal/dependencies/toolsconf/validator.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+func (v *Validator) Validate(s3Conf config.S3) ([]string, []error) {
+	return v.checkAWSS3Bucket(s3Conf.BucketName, s3Conf.Region)
+}
+
+func (v *Validator) checkAWSS3Bucket(bucketName, region string) ([]string, []error) {
+	oks := make([]string, 0)
+	errs := make([]error, 0)
+
+	_, err := v.awsCliRunner.S3("ls", "s3://"+bucketName, "--region", region)
```
Comment (g-iannelli) [inline internal/dependencies/toolsconf/validator.go:44]: This not ensure that bucket region is the right one. 

```
aws s3api get-bucket-location --bucket BUCKET_NAME
```

**Review by ralgozino — APPROVED** [review]
lgtm

File: cmd/create/config.go [hunk 2]
```diff
@@ -81,6 +81,17 @@ func NewConfigCmd(tracker *analytics.Tracker) *cobra.Command {
 				},
 				Spec: distroConfig.FuryctlSpec{
 					DistributionVersion: semver.EnsurePrefix(version),
+					ToolsConfiguration: distroConfig.ToolsConfiguration{
+						Terraform: distroConfig.Terraform{
+							State: distroConfig.State{
+								S3: distroConfig.S3{
+									BucketName: "bucket-name",
+									KeyPrefix:  "key-prefix",
+									Region:     "eu-west-1",
```
Comment (omissis) [inline cmd/create/config.go:90]: Perhaps we shouldn't hard-code this value here?

File: internal/dependencies/tools/validator.go [hunk 3]
```diff
@@ -10,6 +10,7 @@ import (
 	"strings"
 
 	"github.com/sighupio/fury-distribution/pkg/config"
+	tool2 "github.com/sighupio/furyctl/internal/tool"
```
Comment (omissis) [inline internal/dependencies/tools/validator.go:13]:
```suggestion
	itool "github.com/sighupio/furyctl/internal/tool"
```

File: internal/dependencies/tools/validator.go [hunk 4]
```diff
@@ -58,5 +59,14 @@ func (tv *Validator) Validate(kfdManifest config.KFD) ([]string, []error) {
 		}
 	}
 
+	if tfState.S3.BucketName != "" {
+		tool := tv.toolFactory.Create(tool2.Awscli, "*")
```
Comment (omissis) [inline internal/dependencies/tools/validator.go:63]:
```suggestion
		tool := tv.toolFactory.Create(itool.Awscli, "*")
```

File: internal/semver/compare.go [hunk 5]
```diff
@@ -6,150 +6,43 @@ package semver
...
-
-// SamePatch takes two versions and tell if they are part of the same patch.
-func SamePatch(a, b Version) bool {
-	if a == b {
-		return true
-	}
-
-	aMajor, aMinor, aPatch, _ := Parts(a.String())
-	bMajor, bMinor, bPatch, _ := Parts(b.String())
+func GetVersion(v string) (*version.Version, error) {
```
Comment (omissis) [inline internal/semver/compare.go:20]: not a big deal, but creational functions such as this usually follow the `New<thing>` naming pattern as opposed to `Get<thing>`, as they create a new value object (a version in this case): they don't "take it" from somewhere.

File: internal/semver/compare.go [hunk 6]
```diff
@@ -6,150 +6,43 @@ package semver
...
-	if !isValid(pv) {
-		return 0, 0, 0, ""
-	}
-
-	parts := strings.Split(EnsureNoPrefix(v), ".")
-
-	ch := "-"
-	m := strings.Index(v, "-")
-	p := strings.Index(v, "+")
+func GetConstraint(c string) (version.Constraints, error) {
```
Comment (omissis) [inline internal/semver/compare.go:35]: same as per GetVersion above

File: cmd/create/config.go (hunk 2 above)
Comment (Al-Pragliola) [reply cmd/create/config.go:90]: those dummy values are needed by the minimal config validation, we need to think about it in a different card maybe

**Review by omissis — APPROVED** [review]
LGTM

**Pull request merged**